static/src/javascripts/projects/common/modules/identity/account-profile.js

```diff
@@ -96,8 +104,58 @@ define([
         }
     };
 
+    accountProfile.prototype.avatarUploadByGAE = function (avatarForm) {
+        var self = this;
+        var xhr = self.createCORSRequest('GET', self.urls.avatarTokenUrl);
+
+        if (!xhr) {
+            self.prependErrorMessage(self.messages.noCorsError, avatarForm);
+        }
+
+        xhr.onerror = function () {
+            self.prependErrorMessage(self.messages.noServerError, avatarForm);
+        };
+
+        xhr.onload = function () {
+            avatarForm.setAttribute('action', xhr.responseText);
+            avatarForm.submit();
+        };
+
+        xhr.send();
+    };
+
+    accountProfile.prototype.avatarUploadByApi = function (avatarForm) {
+        var self = this;
```

Comment:
Any opportunity to avoid the `var self = this;` pattern here? AFAIK `bind` is safe to use as it's in the [mustard check](https://github.com/guardian/frontend/blob/master/common/app/views/fragments/javaScriptFirstSteps.scala.html#L127).

Not critical, I don't know how common this pattern is in the rest of the codebase.